### PR TITLE
Create Geolocation-Leaflet.html

### LINKED ISSUE #14: 🗺 Geo-location picker
### Motivation

Make it easy for content editors to enter a precise location (latitude and longitude) without needing to know what latitude and longitude is. 

### Proposed solution

Present a **Google maps** address search box and map to allow an editor to search for and refine the selected location. The value of the entered address and lat/long are stored. 

### Resources
* https://developers.google.com/maps/documentation/javascript/examples/places-searchbox

### PR DESCRIPTION
**Pull request by hzik**
### Motivation

See the [description](https://github.com/Kentico/custom-element-samples/blob/bb8053f2fa06039d9f58249ff005530b691e36ca/Geolocation-Leaflet/README.md)

If manual testing is required, what are the steps?
* No, just README.md file.
* Test on the environment.

### CODE REVIEW

**Comment by Simply007** [comment]
Thanks for contribution @hzik!
Could you request the issue first for the next time, because the issue was taken by @ChristopherJennings (he is set as assignee in #14). I would like to prevent this situation for the future.

I will notify him about your pull request.

**Comment by hzik** [comment]
I saw he's assigned to it, but there was Google maps involved, so I've purposely chosen different approach than Google maps, so there can be 2 elements for the same.

**Comment by Simply007** [comment]
Thanks for clarification @hzik!

File: Geolocation-Leaflet/Geolocation-Leaflet.html [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		if (value) {
+			var mapProps = JSON.parse(value);		
+			mymap.setView([mapProps.lat, mapProps.lng], mapProps.zoom);
+			// adds the marker
+			addMarker(mapProps.lat, mapProps.lng);
+		}
+		else {
+			mymap.setView([lat, lng], zoom);
+		}
+		L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoieGhlcm1hbm4iLCJhIjoiY2p0aDB6cHRqMDU5NDRhcDV4YTlmbjh3MiJ9.RS9DeqsnIhaK46H4fQaxVg', {
```
Comment (Simply007) [inline Geolocation-Leaflet/Geolocation-Leaflet.html]: Could you please extract API key definition to the JSON configuration?

It is a good practice to have the ability to configure it from UI.

File: Geolocation-Leaflet/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Geolocation Leaflet element
+
+This KC custom element allows you to put a marker on a map and it automatically gets converted to latitude and longitude geo locations.
+
+This element is based on https://leafletjs.com/ service that uses https://www.openstreetmap.org/ together with the https://www.mapbox.com/ service.
+
+You can test it by configuring https://kentico.github.io/custom-element-samples/Geolocation-Leaflet/Geolocation-Leaflet.html url for your custom element. For testing purposes you can use it as it is (otherwise you need to generate your own MapBox access token - line 70)
```
Comment (Simply007) [inline Geolocation-Leaflet/README.md]: Adjust according to the comment above.

**Comment by hzik** [comment]
- Coordinates added to a marker itself
- Access token extracted :)

**Comment by Simply007** [comment]
Note: added error message when access_key is not provided.

image

**Pull request merged**